climada/util/yearsets.py

```diff
@@ -22,223 +22,227 @@
-    sampling_dict = dict()
-    sampling_dict = {'selected_events': selected_events, 'events_per_year': events_per_year}
 
-    return sampling_dict
+    return events_per_year
 
-def sample_events(tot_n_events, n_input_events):
-    """Sample events (length = tot_n_events) uniformely from an array (n_input_events)
-    without replacement (if tot_n_events > n_input_events the input events are repeated
-                         (tot_n_events/n_input_events) times).
+def sample_events(events_per_year, freqs_orig):
+    """Sample events uniformely from an array (indices_orig) without replacement
+    (if sum(events_per_year) > n_input_events the input events are repeated
+     (tot_n_events/n_input_events) times, by ensuring that the same events doens't
+     occur more than once per sampled year).
 
     Parameters:
-        tot_n_events : int
-            Number of events to be sampled
-        n_input_events : int
-         Number of events contained in given event_impacts object
+        events_per_year : array
+            Number of events per sampled year
+        freqs_orig : array
+            Frequency of each input event
 
     Returns:
-        selected_events : array
-            Uniformaly sampled events (length: tot_n_events)
+        sampling_vect : 2D array
+            The sampling vector specifies how to sample the yimp, it consists of one
+            sub-array per sampled_year, which contains the event_ids of the events used to
+            calculate the annual impacts.
       """
 
-    repetitions = np.ceil(tot_n_events/n_input_events).astype('int')
-    indices = np.tile(np.arange(n_input_events), repetitions)
+    sampling_vect = []
 
-    rng = default_rng()
-    selected_events = rng.choice(indices, size=tot_n_events,
-                                 replace=False).astype('int')
+    indices_orig = np.arange(len(freqs_orig))
 
+    freqs = freqs_orig
+    indices = indices_orig
 
-    return selected_events
+    #if check_doubling true: check that every event doesn't occur more than once per sampled year
+    check_doubling = False
 
-def compute_annual_impacts(event_impacts, sampling_dict):
-    """Sample annual impacts from the given event_impacts according to the sampling dictionary
+    #sample events for each sampled year
+    for idx_year, amount_events in enumerate(events_per_year):
 
-    Parameters:
-        event_impacts : impact object
-            impact object containing impacts per event
-        sampling_dict : dict
-            The sampling dictionary containing two arrays:
-                selected_events (array) : sampled events (len: total amount of sampled events)
-                events_per_year (array) : events per sampled year
+        probab_dis = freqs/np.sum(freqs)
 
-    Returns:
-        impact_per_year: array
-            Sampled impact per year (length = n_sampled_years)
-      """
+        rng = default_rng()
+        selected_events = rng.choice(indices, size=amount_events, replace=False,
```

Comment:
Whre is `amount_events` defined?

Reply:
It's part of the loop (for idx_year, amount_events in enumerate(events_per_year):)
It is the number of events for the particular year being sampled for.
Would n_events be clearer?

Reply:
The name is fine, I just somehow missed where it is defined. Thanks!